smoke/IotEdgeQuickstart/details/Details.cs

```diff
@@ -585,7 +599,8 @@ string BuildImageName(string name)
 
     public class DeviceContext
     {
-        public Device Device;
+        public Option<Device> Device;
+        public string DeviceId;
```

Comment:
Since both Device and DeviceId are public, Device.Id and DeviceId can be different.  I think it is better to make them as properties and make sure to keep them in sync.